src/main/java/com/hazelcast/kubernetes/HazelcastKubernetesDiscoveryStrategy.java

```diff
@@ -104,6 +103,22 @@
         logger.info("Kubernetes Discovery activated resolver: " + endpointResolver.getClass().getSimpleName());
     }
 
+    private String getApiToken(Map<String, Comparable> properties) {
+        String apiToken = getOrDefault(properties, KUBERNETES_SYSTEM_PREFIX, KUBERNETES_API_TOKEN, null);
```

Comment:
I did not check what is the last argument we pass as `null` but It would be good to create overloaded methods if it is easy for easy readability...

Reply:
Good idea! Actually we already have the overloaded method `getOrNull()`. Changed.